platform/openide.util.lookup/test/unit/src/org/openide/util/lookup/ProxyLookupFactoryMethodsTest.java

```diff
@@ -0,0 +1,195 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.openide.util.lookup;
+
+import java.util.Arrays;
+import static java.util.Arrays.asList;
+import java.util.HashSet;
+import java.util.concurrent.CountDownLatch;
+import java.util.concurrent.Executor;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+import java.util.concurrent.ThreadFactory;
+import java.util.concurrent.TimeUnit;
+import java.util.function.BiConsumer;
+import java.util.function.Consumer;
+import org.junit.Test;
+import static org.junit.Assert.*;
+import org.openide.util.Lookup;
+import org.openide.util.LookupEvent;
+import org.openide.util.LookupListener;
+import org.openide.util.lookup.ProxyLookupFactoryMethodsTest.TThreadFactory.TThread;
+
+/**
+ *
+ * @author Tim Boudreau
+ */
+public class ProxyLookupFactoryMethodsTest {
+
+    private Consumer<Lookup[]> lookupConsumer;
+    private BiConsumer<Executor, Lookup[]> lookupBiConsumer;
+
+    private Lookup createWithSingleConsumer(Lookup... lookups) {
+        ProxyLookup.Controller controller = new ProxyLookup.Controller(lookups);
+        lookupConsumer = controller::setLookups;
+        return new ProxyLookup(controller);
```

Comment:
This would be replaced with:
```java
ProxyLookup.Controller controller = new ProxyLookup.Controller();
ProxyLookup l = new ProxyLookup(controller);
l.setLookups(lookups);
return l;
```
I suggest to remove `BiConsumer` altogether as the test should mimic the API usage - e.g. directly use the `Controller`.

Reply:
Are you sure you would not me to make this code compatible with JDK 1.3 while we're at it?  Or perhaps 1.2?

You have something against functional interfaces?  (I am rather [a fan of them](https://github.com/timboudreau/util/tree/master/function/src/main/java/com/mastfrog/function))

As I sit here with Netbeans running on JDK 14, I wonder, why does Jarda want JDK 7 compatibility?

Reply:
I believe the test should show the expected usage of the API which is now without `BiConsumer` & co. Keeping such interface in the implementation or test is a relic of an API which disappeared.

I am running `Lookup` & co. in restricted environments (see for example #2226). Increased usage of JDK8 APIs complicates that.

Reply:
Ok, see if the latest update satisfies your requirements.